### PR TITLE
[WIP] Use Heap for typed arrays.

### PR DESCRIPTION
**Pull request by Ms2ger**
Depends on <https://github.com/servo/rust-mozjs/pull/341>.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15672)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1281,7 +1281,7 @@ dependencies = [
 [[package]]
 name = "js"
 version = "0.1.4"
-source = "git+https://github.com/servo/rust-mozjs#101c6b6b58e0b7c5ed1ef2b6b676a7497f90df2f"
+source = "git+https://github.com/servo/rust-mozjs#25d3ec54c67883aa0a561f5a2f2e485280d113a1"
```
Comment (nox) [inline Cargo.lock]: Please update to tip or to the merge commit of the corresponding PR and r=me.

**Comment by jdm** [comment]
Pretty sure the try push from last time demonstrated some fundamental problems with using RootedTraceableBox for this, since that doesn't actually root nursery-allocated objects.

**Comment by nox** [comment]
http://build.servo.org/builders/linux-rel-wpt/builds/2858

**Comment by Ms2ger** [comment]
Doh.

@bors-servo try

**Comment by Ms2ger** [comment]
For real.

@bors-servo try

**Comment by KiChjang** [comment]
```
Tests with unexpected results:
  ▶ PASS [expected FAIL] /compositing-1_dev/html/mix-blend-mode-animation.htm
```
Is this a good thing?

**Comment by jdm** [comment]
Closing due to inactivity. Tracking in https://github.com/servo/servo/issues/19241.

**Pull request closed without merging**